community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/slurmsync.py

```diff
@@ -472,6 +474,98 @@ def update_topology(lkp: util.Lookup) -> None:
         log.debug("Topology configuration updated. Reconfiguring Slurm.")
         util.scontrol_reconfigure(lkp)
 
+
+def delete_reservations(reservation_map: Dict[str, str]) -> None:
+    slurm_delete_reservation = "scontrol delete reservation {}"
+    for reservation_name in list(reservation_map.keys()):
+      util.run(slurm_delete_reservation.format(reservation_name), timeout=30)
+
+
+def update_slurm_reservation_maintenance(node: str, starttime: str,
+                                         reservation_map: Dict[str, Tuple[str, str]]) -> None:
+    slurm_create_reservation = "scontrol create reservation user=root starttime={} duration=180 nodes={} reservationname={}"
+    slurm_delete_reservation = "scontrol delete reservation {}"
+
+    reservation_name = node + "_maintenance"
+    ## If reservation exists for maintenance, update reservation if time window
+    ## for scheduled maintenance changes.
+    if reservation_name in list(reservation_map.keys()):
```

Comment:
No need to building a `list` here, since you only look it up once.

Reply:
Outdated.